replset-consistency/dbcheck.js

```diff
@@ -154,34 +154,93 @@ function getDBCheckCount(readPref) {
   db.getMongo().setReadPref(readPref);
 
   let curr = db.getSiblingDB("local").system.healthlog.aggregate([
-    {$match : {operation : "dbCheckStart"}}, {$count : "dbCheckStartCount"}
+    { $match: { operation: "dbCheckStart", timestamp: { $gte: startDate } } },
+    { $count: "dbCheckStartCount" },
   ]);
+
   if (curr.hasNext()) {
     let my_count = curr.next().dbCheckStartCount;
     return my_count;
   }
   return 0;
 }
 
-// Non-deterministic, we don't know if we'll select the same secondary over and
-// over again.
+function getDBCheckCountByNode(node) {
+  let conn = node.connection;
+  let curr = conn.getDB("local").getCollection("system.healthlog").aggregate([
+    { $match: { operation: "dbCheckStart", timestamp: { $gte: startDate } } },
+    { $count: "dbCheckStartCount" },
+  ]);
+
+  if (curr.hasNext()) {
+    let my_count = curr.next().dbCheckStartCount;
+    node.dbCheckStartCount = my_count;
+    return my_count;
+  }
+  return 0;
+}
+
+// Authenticates into every data-bearing secondary nodes to query healthlog 
+// dbCheckStart entries
+//
+// If authInfo is not provided, behavior defaults to old healthlog rollover check 
```

Comment:
Let's document this at the very beginning as well, as part of the Usage comment. That is, let's ensure our main usage example recommends an invocation that performs the strong check and warns against running without authInfo.

Let's also include an equally appropriate change to https://github.com/mongodb/support-tools/tree/replset-consistency/replset-consistency/README.md.

Reply:
Added this information in both README and top of dbcheck.js